timesketch/frontend-ng/src/components/Explore/AggregateDialog.vue

```diff
@@ -0,0 +1,399 @@
+    },
+    valueDateTimeMinimum() {
+      if (this.stats === undefined) return null
+      return this.stats.datetime_min.value_as_string
+    },
+    valueDateTimeMaximum() {
+      if (this.stats === undefined) return null
+      return this.stats.datetime_max.value_as_string
+    },
+    fieldValueCount() {
+      if (this.stats === undefined) return null
+      return this.stats.all_values.field_count.value
+    },
+    valueEventCount() {
+      if (this.stats === undefined) return null
+      return this.stats.value_count.value
+    },
+    commonValues() {
+      if (this.stats === undefined) return []
+      return this.stats.all_values.top_terms.buckets
+    },
+    rareValues() {
+      if (this.stats === undefined) return []
+      return this.stats.all_values.rare_terms.buckets
+    },
+    donutChartOptions() {
+      return {
+        chart: {
+          type: 'donut'
+        },
+        dataLabels: {
+          enabled: true
+        },
+        labels: [this.eventValue, 'Other ' + this.eventKey],
+      }
+    },
+    donutChartSeries() {
+      return [this.valueEventCount, this.fieldValueCount - this.valueEventCount]
+    },
+    minEvent: function() { return this.raw_data[0] },
+    maxEvent: function() { return this.raw_data[this.raw_data.length - 1] },
+    intervalHeatmapOptions() {
+      return {
+        chart: {
+          type: 'heatmap',
+        },
+        colors: ["#008FFB"],
+        tooltip: {
+          enabled: true,
+          x: { show: true },
+        },
+        xaxis: {
+          labels: { show: true, hideOverlappingLabels: true },
+          tickPlacement: "between",
+        },
+      }
+    },
+    intervalHeatmapSeries() {
+      let series = []
+
+      if (this.stats === undefined) return series
+
+      for (let day of Array.from({ length: 7}).keys()) {
+        let daySeries = []
+        console.log(day)
+        for (let hour of Array.from({ length: 24}).keys()) {
+          const count = this.stats.hour_of_week_histogram.buckets[day*24 + hour].doc_count
+          console.log(day, hour, count)
+          daySeries.push({x: this.hoursOfDay[hour], y: count})
+        }
+        series.push({ 'name': this.daysOfWeek[day], 'data': daySeries})
+      }
+      console.log(series)
+      return series
+    },
+  },
+  watch: {
+    eventKey (value, oldValue) {
+      if (value !== oldValue) {
+          this.loadSummaryData()
+        }
+    },
+    eventValue (value, oldValue){
+      if (value !== oldValue) {
+          this.loadSummaryData()
+        }
+    },
+  },
+  methods: {
+    loadSummaryData: function() {
+      this.stats = undefined
+      this.raw_data = {}
+      ApiClient.runAggregator(this.sketchId, {
+        aggregator_name: 'field_summary',
+        aggregator_parameters: {
+          field: this.eventKey,
+          field_query_string: this.eventValue
+        }
+      }).then((response) => {
+        this.stats = response.data.objects[0].field_summary.buckets[0]
```

Comment:
During testing your PR I requarly got the following error on the console when loading aggregations for various fields:
```
Uncaught (in promise) TypeError: Cannot read properties of null (reading 'type')
```
My theory on this problem: It has to do with some race condition where `stats` is not yet defined by the mounted method call and the graphs already trying to call the computed methods. 

On way to prevent this race condition that I can think of:
1. Add a new variable `statsReady: false`to the data above.
2. Set this variable to true when the promise here is fulfilled.
3. Add the `:loading="!statsReady"` option to the `v-card` elements to display a loading animation when the data is not ready yet. 
4. Add `v-if="statsReady"` to the `v-card-text` elements that contain the apexchart or to the apexchart itself to only render it when the data is available.

At least in my tests, this did prevent all errors and also shows a nice loading animation with aggregations on larger data sets.

Reply:
Thanks I have implemented the proposed solution and it seems to work nicely.